Modules/zlibmodule.c

```diff
@@ -38,10 +38,12 @@ Buffer_InitWithSize(_BlocksOutputBuffer *buffer, Py_ssize_t init_size,
 {
     Py_ssize_t allocated;
 
-    if (init_size < 0 || (size_t)init_size > UINT32_MAX) {
-        PyErr_SetString(PyExc_ValueError,
-                        "Initial buffer size should (0 <= size <= UINT32_MAX)");
-        return -1;
+    if (init_size >= 0 &&  // ensure (size_t) cast is safe
+        (size_t)init_size > UINT32_MAX)
+    {
+        /* In 32-bit build, never reach this conditional branch.
+           The maximum block size accepted by zlib is UINT32_MAX. */
+        init_size = UINT32_MAX;
     }
 
```

Comment:
No longer raise `ValueError` for `(init_size < 0)`, both the previous code [1] and the following code [2] check this.

[1] the previous code
https://github.com/python/cpython/blob/f9bedb630e8a0b7d94e1c7e609b20dfaa2b22231/Modules/zlibmodule.c#L348-L350
https://github.com/python/cpython/blob/f9bedb630e8a0b7d94e1c7e609b20dfaa2b22231/Modules/zlibmodule.c#L1178-L1181

[2] the following code
https://github.com/python/cpython/blob/f9bedb630e8a0b7d94e1c7e609b20dfaa2b22231/Objects/bytesobject.c#L139-L146

Reply:
Good catch!